Lib/typing.py

```diff
@@ -725,7 +726,11 @@ def __ror__(self, other):
         return Union[other, self]
 
     def __repr__(self):
-        return f'ForwardRef({self.__forward_arg__!r})'
+        if self.__forward_module__ is None:
+            module_repr = ''
+        else:
+            module_repr = f', module={self.__forward_module__!r}'
+        return f'ForwardRef({self.__forward_arg__!r}{module_repr})'
```

Comment:
This repr doesn't feel too consistent with the rest of Python. IMO, we should have the module before the forward arg. Eg.:
```python
>>> class X: ...
...
>>> X
<class '__main__.X'>
```
In the above example, the format is `{module}.{classname}`. I feel that we should follow that.

Reply:
I see your point. However, this is how `ForwardRef` works - it has the extra `module` argument. So repr corresponds to the constructor.

Presently you cannot do `ForwardRef('mod.YY')` as this would not resolve correctly (Only if you have 'mod' imported in the current workspace, and also only if `globalsns` is set during `_evaluate`, which defeats the purpose of having the `module` argument in the first place).  

But I agree, this would be a extremely nice thing to have.  It would allow for a much terser usage `Optional[__name__ + '.YY']` is way more readable than `Optional[ForwardRef('YY', module=__main__)]`.

Reply:
Sorry, the repr idea was based off a misconception I had in https://github.com/python/cpython/pull/30536#discussion_r783093807. I'll post some related concerns on the bug tracker. Thank you for your well-thought out response!

Reply:
@Fidget-Spinner I added some background info in the bug tracker issue.